prime-router/docs/design/proposals/0025-handling-multiple-specs/0025-handling-multiple-specs.md

```diff
@@ -0,0 +1,189 @@
+
+This includes:
+
+- The structure of the groups of the segments
+- The shape of the individual segments
+- Specialized version of data types
+    - i.e. the NIST ELR 2.5.1 conformance profile defines a `CWE_ELR` datatype
+- Any constraints
+    - i.e. if a field is repeatable
+
+There are two different versions of conformance profiles (samples of which are in this directory):
+
+- one is supported by the HAPI Source gen tool
+- one can be constructed by the NIST IGAMT tool
+
+At a high level, both versions can be referenced in order to create a HAPI java message that specifically implements the
+profile.
+
+## Goal
+
+The goal is to create a system that can deterministically handle mapping from HL7->FHIR->HL7 when ReportStream
+knows that the message lines up with a defined conformance profile allowing us to guarantee senders/receivers a lossless
+translation.
+
+## Approach
+
+### Short term handling for new conformance profiles
+
+The overall proposal is to ask senders to use `MSH.21` to indicate which conformance profile they are targeting. If
+ReportStream can not determine an implemented conformance profile, it will continue to fallback to comprehensive
+HL7->FHIR->HL7 mappings. This means that receivers can continue to get their messages with a high degree of fidelity
+just without the absolute guarantee of all the data making it through.
+
+For each conformance profile ReportStream decides to support, the following steps are taken:
+
+1. Create an implementation of `AbstractMessage` that matches the conformance profile.
+    - The HAPI HL7 library provides a tool for generating these from a conformance profile. See the section below for
+      more details
+2. Copy the generic HL7->FHIR mappings based on the FHIR inventory and then update them to exactly reflect what the
+   conformance profile specifies.
+    - This work could entail removing schemas for deprecated fields or creating new custom data types (i.e. the radxmars
+      spec defines a `CWE_NIH` type)
+3. Extend the generic FHIR -> HL7 mapping and update it to reflect the conformance profile
+    - That work would entail disabling certain elements if the field is deprecated or mapping custom datatypes
+4. Create new tests for each segment and custom data type
+
+### Converting between specs
+
+One additional complexity to consider is how to convert between different conformance profiles; imagine a sender with
+NIST ELR 2.5.1 messages and one receiver that is getting NIST ELR 2.5.1 and another that is receiving radxmars. The
+actual mechanics of how to do this would need to be handled after a comparison of the two conformance profiles occurs,
+but once that comparison is done FHIR transforms can be written to modify the bundle, so it can be translated
+successfully to a different conformance profile.
+
+### Rough code updates
+
+- Update the `HL7toFhirTranslator` to be instantiated with the templates that reflect the conformance profile
```

Comment:
- Update the `HL7toFhirTranslator` to be instantiated with the templates that reflect the conformance profile. If a profile is not provided, default to `what?`

Reply:
I'm actually going to pull this section out since it's now captured in the tickets, but we'll have a directory of templates called `fallback` that will be used if a profile cannot be determined because it's not provided or reportstream has not implemented it.

I'll make sure that is clearer in the doc too.